### PR TITLE
GraphNG: support x != time from the UI

### PR DESCRIPTION
**Pull request by ryantxu**
This explores how we can support x != time, and starts to explore a more generic dimension mapping approach

image

Test with "random walk table" and pick min vs max field

### CODE REVIEW

**Comment by leeoniya** [comment]
probably hold off on getting too deep into this until proper scatter support lands in uPlot: https://github.com/leeoniya/uPlot/issues/107

**Comment by torkelo** [comment]
nice! exciting to get to this! 

would have been good to base it on https://github.com/grafana/grafana/pull/29275, going to be messy to merge. Been fixing a couple of graph issues in that PR as well (time axis, points mode, pass min & max)

**Comment by ryantxu** [comment]
The interesting bits are all outside GraphNG/react, so I am not too worried about the merge.

**Comment by leeoniya** [comment]
thinking a bit beyond basic x/y scatter...would something like below (multi-series bubble chart) be possible to extract from a single DataFrame? you'd probably need to do something like a field-chunker (take every 4 fields).

```js
[
  [  // series 1
     [1,2,3],  // x
     [10,20,30],  // y
     [2.2,1.5,6.5], // v
     ["a","b","c"],  // l
  ],
  [  // series 2
     [1,2,3],  // x
     [10,20,30],  // y
     [2.2,1.5,6.5], // v
     ["a","b","c"],  // l
  ]
]
```

**Comment by ryantxu** [comment]
> possible to extract from a single DataFrame?

um... we start with `DataFrame[]` that has everything you need :)

File: packages/grafana-data/src/transformations/matchers.ts [hunk 1]
```diff
@@ -30,16 +32,10 @@ export const frameMatchers = new Registry<FrameMatcherInfo>(() => {
 
 export function getFieldMatcher(config: MatcherConfig): FieldMatcher {
   const info = fieldMatchers.get(config.id);
-  if (!info) {
```
Comment (ryantxu) [inline packages/grafana-data/src/transformations/matchers.ts:33]: not required since the `.get` above will throw an error

**Comment by ryantxu** [comment]
closed by #30096

**Pull request closed without merging**